File: Chat.Api/ChatService/Queries/ChatListQuery.cs
using Chat.Api.ChatService.Interfaces;
using Chat.Api.Core.Models;
using Chat.Api.Core.Services;

namespace Chat.Api.ChatService.Queries
{
    public class ChatListQuery : AQuery
    {
        public string UserId {get; set;} = string.Empty;

        public override void ValidateQuery()
        {
            if (string.IsNullOrEmpty(UserId))
            {
                throw new Exception("UserId not set");
            }
        }
    }
}

File: chat-app/src/app/chat/queries/chat-list-query.ts
import { QueryBase } from "src/app/core/models/query-base";
import { Configuration } from "src/app/core/services/configuration";

export class ChatListQuery extends QueryBase {
    
    userId : any;
    
    constructor() {
        super();
        this.apiUrl = Configuration.identityApi + "/chat/list";
        this.Offset = 0;
        this.limit = 100;
    }
}